packages/sanity/src/core/versions/components/panes/BundleActions.tsx

```diff
@@ -1,32 +1,45 @@
-import {AddIcon} from '@sanity/icons'
+import {AddIcon, CheckmarkIcon} from '@sanity/icons'
 import {useToast} from '@sanity/ui'
 import {useCallback, useEffect, useState} from 'react'
-import {DEFAULT_STUDIO_CLIENT_OPTIONS, useClient, useDocumentOperation} from 'sanity'
+import {filter, firstValueFrom} from 'rxjs'
+import {
+  DEFAULT_STUDIO_CLIENT_OPTIONS,
+  useClient,
+  useDocumentOperation,
+  useDocumentStore,
+} from 'sanity'
 
 import {Button} from '../../../../ui-components'
-import {type Version} from '../../types'
+import {useBundles} from '../../../store/bundles/BundlesProvider'
+import {type BundleDocument} from '../../../store/bundles/types'
 import {getAllVersionsOfDocument, versionDocumentExists} from '../../util/dummyGetters'
 
 interface BundleActionsProps {
-  currentVersion: Version
+  currentVersion: BundleDocument
   documentId: string
   documentType: string
-  isReady: boolean
 }
 
 export function BundleActions(props: BundleActionsProps): JSX.Element {
-  const {currentVersion, documentId, documentType, isReady} = props
+  const {currentVersion, documentId, documentType} = props
   const {name, title} = currentVersion
+  const {data: bundles, loading} = useBundles()
+  const documentStore = useDocumentStore()
 
-  const [documentVersions, setDocumentVersions] = useState<Version[]>([])
+  const [documentVersions, setDocumentVersions] = useState<BundleDocument[]>([])
+  const [creatingVersion, setCreatingVersion] = useState(false)
+  const [isInVersion, setIsInVersion] = useState(false)
   const client = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
   const toast = useToast()
   const {newVersion} = useDocumentOperation(documentId, documentType)
 
   const fetchVersions = useCallback(async () => {
-    const response = await getAllVersionsOfDocument(client, documentId)
-    setDocumentVersions(response)
-  }, [client, documentId])
+    if (!loading) {
+      const response = await getAllVersionsOfDocument(bundles, client, documentId)
+      setDocumentVersions(response)
+      setIsInVersion(versionDocumentExists(documentVersions, name))
+    }
```

Comment:
we will eventually get this from the document pair store right? 
cc: @juice49

Reply:
We've made a decision to move the discussion at a later point once we have a bit more information